src/fees/compute_fees.py

```diff
@@ -62,20 +84,38 @@ def surplus(self) -> int:
             return current_limit_sell_amount - self.sell_amount
         raise ValueError(f"Order kind {self.kind} is invalid.")
 
-    def raw_surplus(self) -> int:
-        """Compute raw surplus of a trade in the surplus token
-        First, the application of protocol fees is reversed. Then, surplus of the resulting trade
-        is computed."""
+    def compute_all_fees(self) -> None:
         raw_trade = deepcopy(self)
-        for fee_policy in reversed(self.fee_policies):
-            raw_trade = fee_policy.reverse_protocol_fee(raw_trade)
-        return raw_trade.surplus()
+        i = 0
+        self.protocol_fee = 0
+        self.partner_fee = 0
+        if self.fee_policies != []:
+            for fee_policy in reversed(self.fee_policies):
+                raw_trade = fee_policy.reverse_protocol_fee(raw_trade)
+                ## we assume that partner fee is the last to be applied
+                if i == 0 and self.partner_fee_recipient is not NULL_ADDRESS:
+                    self.partner_fee = raw_trade.surplus() - self.surplus()
+                i = i + 1
+            self.protocol_fee = raw_trade.surplus() - self.surplus() - self.partner_fee
```

Comment:
This deviates from the convention we use elsewhere. There, protocol fees are always the sum of all protocol fee (even those designated for integrators).

If you think we should change the convention, we should change it elsewhere as well. Or document the different meanings in different places.

Reply:
Oh good point. I always need to double-check to remind myself of the convention, so better stick with what we have been doing so far. Just pushed a change, but also renamed it to `total_protocol_fee` to make it a bit more explicit